capstone/src/main/webapp/youtubeScript.js

```diff
@@ -23,29 +23,41 @@ const ATTRIBUTES_BY_LANGUAGE = {
 
 /** Calls youtube servlet and passes output to perspctive */
 async function callYoutube() {
-  const channelId = document.getElementById('channelIdForAnalysis').value;
+  const channelId = document.getElementById('channelIdForAnalysis').value.replace(/ /g, '');
   if (!channelId) {
     return;
   }
   var response;
-  if (channelId[0] == "U" && channelId[1] == "C" && channelId.length == 24) {
+  if (channelId[0] == "U" && channelId[1] == "C" && channelId.length == 24 && isLetter(channelId[channelId.length-1])) {
     response = await fetch('/youtube_servlet?channelId=' + channelId,);
+    response = await response.json();
+    if (response.hasOwnProperty('error')) {
+      alert("Invalid Channel ID");
+      inputCommentsToPerspective([]);
+      return;
+    }
   } else {
     const converterResponse = await fetch('/username_servlet?channelId=' + channelId,)
```

Comment:
nit: The `fetch()` calls here and on line 48 have a comma at the end that should not be there

Reply:
explain channel ID manipulation and remove semicolons